app/src/main/java/com/kickstarter/ui/views/CommentCard.kt

```diff
@@ -0,0 +1,151 @@
+                OnClickListener {
+                    onCommentCardClickedListener?.onCommentGuideLinesClicked(it)
+                },
+
+            ),
+            linkColor = R.color.kds_create_500
+        )
+    }
+
+    private fun obtainStyledAttributes(context: Context, attrs: AttributeSet?, defStyleAttr: Int) {
+        context.withStyledAttributes(
+            set = attrs,
+            attrs = R.styleable.CommentCardView,
+            defStyleAttr = defStyleAttr
+        ) {
+            getString(R.styleable.CommentCardView_comment_card_message)?.also {
+                setCommentBody(it)
+            }
+            getString(R.styleable.CommentCardView_comment_card_post_time)?.also {
+                setCommentPostTime(it)
+            }
+            getString(R.styleable.CommentCardView_comment_card_avatar_url)?.also {
+                setAvatarUrl(it)
+            }
+
+            getString(R.styleable.CommentCardView_comment_card_user_name)?.also {
+                setCommentUserName(it)
+            }
+            getInt(R.styleable.CommentCardView_comment_card_status, 0).also { attrValue ->
+                CommentCardStatus.values().firstOrNull {
+                    it.commentCardStatus == attrValue
+                }?.let {
+                    setCommentCardStatus(it)
+                }
+            }
+        }
+    }
+
+    fun setCommentCardStatus(cardCommentStatus: CommentCardStatus) {
+        binding.commentDeletedMessageGroup.isVisible =
+            cardCommentStatus == CommentCardStatus.DELETED_COMMENT
+
+        binding.commentBody.isVisible = cardCommentStatus == CommentCardStatus.COMMENT_WITH_REPLAY ||
+            cardCommentStatus == CommentCardStatus.COMMENT_WITHOUT_REPLAY
+
+        binding.commentActionGroup.isVisible = cardCommentStatus == CommentCardStatus.COMMENT_WITH_REPLAY ||
+            cardCommentStatus == CommentCardStatus.COMMENT_WITHOUT_REPLAY
+
+        binding.retryButton.isVisible =
+            cardCommentStatus == CommentCardStatus.FAILED_TO_SEND_COMMENT
+
+        val commentBodyTextColor = if (cardCommentStatus == CommentCardStatus.FAILED_TO_SEND_COMMENT) {
+            R.color.soft_grey_disable
+        } else {
+            R.color.text_primary
+        }
+
+        binding.commentBody.setTextColor(ContextCompat.getColor(context, commentBodyTextColor))
+    }
+
+    fun setCommentUserName(username: String) {
+        binding.commentUserName.text = username
+    }
+
+    fun setCommentPostTime(time: String) {
+        binding.commentPostTime.text = time
+    }
+
+    fun setCommentBody(comment: String) {
+        binding.commentBody.text = comment
+    }
+
+    fun setCommentCardClickedListener(onCommentCardClickedListener: OnCommentCardClickedListener?) {
+        this.onCommentCardClickedListener = onCommentCardClickedListener
+    }
+
+    fun setAvatarUrl(url: String?) {
+        url?.let {
+            Picasso.get().load(it)
```

Comment:
This is a picky comment for the future, I would like to not speak directly to a dependency, this comment is more meant for future tech debt as I know this concrete case with picasso is quite extended throwout the app so no need to act on it.
But we should create or a utils class or a client to inject on the UI layer that will isolate us from calling any third party directly

Reply:
we can also create an extension layer or Custom Imageview using through app

Reply:
totally agree

Reply:
![image](https://user-images.githubusercontent.com/1075310/119357321-b5c3cb00-bca7-11eb-8d94-58aef6645407.png)
![image](https://user-images.githubusercontent.com/1075310/119357470-e6a40000-bca7-11eb-82a1-a47c6e23531f.png)